### PR TITLE
v0.3.38

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Changelog -

- [Universal] - Updates `git` to v2.45.1 to get the latest security bugfix release. See https://github.blog/2024-05-14-securing-git-addressing-5-new-vulnerabilities/
- [Universal] - Pins .NET Runtime & Core versions to `7.0.18` to unblock from flakey dotnet-install script. See [here](https://github.com/devcontainers/images/pull/1065/files#diff-0a764bd0944b79c46597d08357a72f308324874b8e25e7d46f302ecc0bc47bafR140a764bd0944b79c46597d08357a72f308324874b8e25e7d46f302ecc0bc47bafR14)
- [[Universal] - reduce size - removed nvidia, triton - Machine Learning Packages](https://github.com/devcontainers/images/pull/1014)
- [[Anaconda]-Werkzeug-GHSA-2g68-c3qc-8985 patch for security vuln](https://github.com/devcontainers/images/pull/1062)
- [[Anaconda]-Jinja2-GHSA-h75v-3vvj-5mfj patch security vuln](https://github.com/devcontainers/images/pull/1063)

### CODE REVIEW

File: src/universal/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -11,8 +11,10 @@
         },
         "ghcr.io/devcontainers/features/dotnet:2": {
             "version": "8.0",
-            "dotnetRuntimeVersions": "7.0",
-            "aspNetCoreRuntimeVersions": "7.0"
+			// Temporarily pinning as the dotnet-install script is flakey (attempts to install bits of 7.0.19 of Runtime, but 7.0.18 of Core & Runtime)
+			// See https://github.com/devcontainers/images/actions/runs/9083566665/job/24962519615?pr=1065#step:3:6146
```
Comment (samruddhikhandale) [inline src/universal/.devcontainer/devcontainer.json:15]: Opened https://github.com/devcontainers/images/issues/1066 for tracking unpinning

File: src/universal/test-project/test.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ checkCommon
 check "git" git --version
 
 git_version=$(git --version)
-check-version-ge "git-requirement" "${git_version}" "git version 2.40.1"
+check-version-ge "git-requirement" "${git_version}" "git version 2.45.1"
```
Comment (cmbrose) [inline src/universal/test-project/test.sh:12]: Just for my own understanding - this is just verifying the version, but where are we pinning it? Or is the upstream already updated and we didn't need to pin?

**Review by joshspicer — APPROVED** [review]

File: src/universal/test-project/test.sh (hunk 2 above)
Comment (samruddhikhandale) [reply src/universal/test-project/test.sh:12]: We install `git` via the Git Feature, see 👇 
 https://github.com/devcontainers/images/blob/3dd43c5939df25075315c737b81cc167ac94abb8/src/universal/.devcontainer/devcontainer.json#L56-L59
 
With `ppa:false`, it always installs from source and with `version:latest`, it always installs latest. Hence, we don't need any explicit changes! ✨

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
Need support

**Pull request merged**